### ISSUE TITLE
[OLD] Legacy Arabic Translation Progress

### ISSUE THREAD

**Issue by tesseralis** [reactjs/ar.react.dev#1]
## Maintainer List

* @asantarissy 
* @iRayan7 
* @aladin002dz
* @Fcmam5 
* @mohshbool 
* @3imed-jaberi 
* @splimter 

## For New Translators

To translate a page:

1. Check that no one else has claimed your page in the checklist and comments below.
2. Comment below with the name of the page you would like to translate. **Please take only one page at a time**.
3. Clone this repo, translate your page, and submit a pull request!

Before contributing, read the [glossary](https://github.com/reactjs/ar.reactjs.org/wiki/Glossary) and [style guide](https://github.com/reactjs/reactjs.org-translation/blob/master/style-guide.md) to understand how to translate various technical and React-specific terms.

Please be prompt with your translations! If you find that you can't commit any more, let the maintainers know so they can assign the page to someone else.

PS: Please use [Hsoub's wiki](https://wiki.hsoub.com/React) as your reference but keep in your mind we need to keep the tech. terms in the English and Arabic language.
eg: unit testing (وحدات الاختبار), javascript (جافاسكريبت),...etc.

## For Maintainers

When someone volunteers, edit this issue with the username of the volunteer, and with the PR. Ex:

* [x] Home Page (@tesseralis) #1

When PRs are merged, make sure to mark that page as completed!

* [x] RTL Support (@asantarissy) #8

## Core Pages

To do before releasing as an "official" translation. **Please translate these pages first.**

* [x] Home Page (@fkanout) #36, #37, #38 & #39 
* [x] Tutorial (@3imed-jaberi) #135 

### Basics

* [x] Hello World (@maskeddeveloper) #125 
* [x] Introducing JSX (@asantarissy) #9 
* [x] Rendering Elements (@omar-dulaimi) #10 
* [x] Components and Props (@asantarissy) #11 
* [x] State and Lifecycle (@3imed-jaberi) #115 
* [x] Handling Events (@aladin002dz) #21 
* [x] Conditional Rendering (@AdhamMoussa) #22 
* [x] Lists and Keys (@splimter) #53 
* [x] Forms (@aladin002dz) #17 
* [x] Lifting State Up (@aladin002dz) #25 
* [x] Composition vs Inheritance (@a-deeb) #29
* [x] Thinking in React (@ahmedfarag36) #31 

### API Reference

* [x] React (@hgareeballa) #32 
* [x] React.Component (@splimter) #40
* [x] ReactDOM (@splimter) #43
* [x] ReactDOMServer (@splimter) #71 
* [x] DOM Elements (@splimter) #44
* [x] SyntheticEvent (@splimter) #45
* [x] Test Utilities (@splimter) #46
* [x] Shallow Renderer (@splimter) #47
* [x] Test Renderer (@splimter) #48
* [x] JS Environment Requirements (@splimter) #49
* [x] Glossary (@splimter) #51

### Navigation

These are the navigation links that appears in the sidebar. Possibly wait until
the corresponding sections are translated to do these.

* [x] tutorial/nav.yml (@3imed-jaberi ) #136 
* [x] docs/nav.yml (@3imed-jaberi ) #97 

## Next Steps

These are the next most important translations, ordered by importance:

### Installation

* [x] Getting Started (@mhmdAljefri) #54
* [x] Add React to a Website (@mhmdAljefri) #55 
* [x] Create a New React App (@mhmdAljefri) #56
* [x] CDN Links (@mohshbool) #70 
* [x] Release Channels (@mohamedsgap) #205

### Hooks

Everyone is hooked on hooks! People are really excited for these and it would be great for them to be translated.

  * [x] Introducing Hooks (@splimter) #57 
  * [x] Hooks at a Glance (@splimter) #58 
  * [x] Using the State Hook (@splimter) #59 
  * [x] Using the Effect Hook (@aladin002dz) #60 
  * [x] Rules of Hooks (@splimter) #61 
  * [x] Building Your Own Hooks (@splimter) #63 
  * [x] Hooks API Reference (@splimter) #68  
  * [x] Hooks FAQ (@splimter) #69 

### Testing
* [x] Testing Overview (@mohdule) #195 
* [x] Testing Recipes (@mohamedsgap) #198 
* [x] Testing Environments (@mohamedsgap) #202 

### Concurrent Mode ( New ) 
* [ ] Introducing Concurrent Mode (@Fcmam5) #207
* [x] Suspense for Data Fetching (@mohamedsgap) #208 
* [ ] Concurrent UI Patterns (@yusuf963) #454 
* [x] Adopting Concurrent Mode (@mohamedsgap) #218 
* [x] Concurrent Mode API Reference (@mohamedsgap) #225  (@3imed-jaberi) #243 

### Warnings

These are the pages that you get when you click the links in the console (e.g. https://reactjs.org/warnings/dont-call-proptypes.html). People tend to search these, so **please don't translate the error message itself**.
 
* [x] dont-call-proptypes (@mohshbool) #121 
* [x] invalid-aria-prop (@mohshbool) #123 
* [x] invalid-hook-call-warning (@mohshbool) #126 
* [x] legacy-factories (@mohshbool) #145 
* [x] refs-must-have-owner (@mohshbool) #134
* [x] special-props (@mohshbool) #141
* [x] unknown-prop (@mohshbool) #132 

### Advanced Guides

* [x] Accessibility (@splimter) #64
* [x] Code-Splitting (@splimter) #65
* [x] Context (@splimter) #66
* [x] Error Boundaries (@aminebenkeroum) #67 
* [x] Forwarding Refs (@mohshbool) #91 
* [x] Fragments (@splimter) #74 
* [x] Higher-Order Components (@aladin002dz) #119 
* [x] Integrating with Other Libraries (@3imed-jaberi) #102 
* [x] JSX In Depth (@3imed-jaberi) #103 
* [x] Optimizing Performance (@fcmam5) #96 
* [x] Portals (@3imed-jaberi) #101 
* [x] Profiler API (@ahmed-abdulhuy) #395 
* [x] React Without ES6 (@aladin002dz) #93
* [x] React Without JSX (@m1m6) #78 
* [x] Reconciliation (@Fcmam5) #99 
* [x] Refs and the DOM (@mohshbool) #108 
* [x] Render Props (@Fcmam5) #105 
* [x] Static Type Checking (@3imed-jaberi) #89 
* [x] Strict Mode (@3imed-jaberi) #92 
* [x] Typechecking With PropTypes (@AbdelkhalekESI) #80
* [x] Uncontrolled Components (@3imed-jaberi) #95 
* [x] Web Components (@3imed-jaberi) #94 
* [x] Web Components  (@3imed-jaberi) #215

### FAQ

* [x] APIs and AJAX (@maskeddeveloper) #83 
* [x] Babel, JSX, and Build Steps (@3imed-jaberi) #138 
* [x] Passing Functions to Components (@3imed-jaberi) #139 
* [x] Component State (@maskeddeveloper) #85 
* [x] Styling and CSS (@maskeddeveloper) #86
* [x] File Structure (@maskeddeveloper) #88
* [x] Versioning Policy (@maskeddeveloper) #116  
* [ ] Versioning Policy [#minors-versus-patches](https://ar.reactjs.org/docs/faq-versioning.html?no-cache=1#minors-versus-patches)
* [x] Virtual DOM and Internals (@3imed-jaberi ) #114 

### Contributing

Priority: Low

* [x] How to Contribute (@Fcmam5) #120 
* [ ] How to Contribute (@ahmed-abdulhuy) [#branch-organization](https://ar.reactjs.org/docs/how-to-contribute.html#branch-organization) & [#feature-flags](https://ar.reactjs.org/docs/how-to-contribute.html#feature-flags) #394
* [x] Codebase Overview (@ahmedkrmn) #131
* [ ] Implementation Notes (@mriiad) #372 
* [ ] Design Principles

### Components

Components in `src/components` that have some text in them.

* [x] CodeEditor (@Fcmam5) #110 
* [x] CodeExample (@Fcmam5) #110 
* [x] ErrorDecoder (@Fcmam5) #110 
* [x] LayoutFooter (@Fcmam5) #110 
* [x] LayoutHeader (@Fcmam5) #110 
* [x] MarkdownPage (@Fcmam5) #110 

## Additional Translations

These are not the primary translation targets.

* [ ] Blog
* [x] Community #117
* [x] Community - Meetups (@3imed-jaberi) #221 
* [ ] Community - Team

**Comment by tesseralis** [comment]
@22mahmoud, @SaifAsad, @Emmanuel-Melon, @iRayan7 how's it going? Do you have any questions or need any help getting started?

**Comment by fawzisf** [comment]
I would love to help! i see no pages are currently taken so i will start by the home page.

**Comment by Emmanuel-Melon** [comment]
> @22mahmoud, @SaifAsad, @Emmanuel-Melon, @iRayan7 how's it going? Do you have any questions or need any help getting started?

Yes, I do. Which repository should I clone and push changes to? Is it [https://github.com/reactjs/ar.reactjs.org](url) or [https://github.com/reactjs/reactjs.org](url)?

**Comment by fawzisf** [comment]
@Emmanuel-Melon This one.

**Comment by Emmanuel-Melon** [comment]
> @Emmanuel-Melon This one.

Thanks, I have just cloned it. BTW, have you started working on the home page?

**Comment by fawzisf** [comment]
No, but willing to start today.

**Comment by Emmanuel-Melon** [comment]
> No, but willing to start today.

Make a pull request so I could attach your name to the page you're editing.

**Comment by fawzisf** [comment]
@tesseralis i think we will need some help as well to transform the website to RTL

**Comment by tesseralis** [comment]
@fawzisf I'll look into it! For now, maybe looking at what the Persian translation is doing might be useful: https://github.com/reactjs/fa.reactjs.org/pull/7/files

**Comment by IsraaMoqbel** [comment]
Hey,
I'm interested to contribute to this, I'll do the `Hello world` page.

**Comment by omar-dulaimi** [comment]
Can I join you guys please? 
BTW "unit testing" = "اختبار الوحدات"

**Comment by 22mahmoud** [comment]
Hey @IsraaMoqbel , Please open a PR, So I could assign you to this page

**Comment by asantarissy** [comment]
> Can I join you guys please?
> BTW "unit testing" = "اختبار الوحدات"

Thanks @omar-dulaimi, please read [The Contribution Guidelines](https://github.com/reactjs/ar.reactjs.org#contributing), pick a page to translate and open a PR so I can assign you to it. enjoy!

الله حيو النشمي 😄

**Comment by omar-dulaimi** [comment]
> > Can I join you guys please?
> > BTW "unit testing" = "اختبار الوحدات"
> 
> Thanks @omar-dulaimi, please read [The Contribution Guidelines](https://github.com/reactjs/ar.reactjs.org#contributing), pick a page to translate and open a PR so I can assign you to it. enjoy!
> 
> الله حيو النشمي 😄

Thanks man @tesseralis , I have read them all just a while ago. I'm going to pick 'rendering elements' if I may.
الله يحييك  😁

**Comment by asantarissy** [comment]
> > > Can I join you guys please?
> > > BTW "unit testing" = "اختبار الوحدات"
> > 
> > 
> > Thanks @omar-dulaimi, please read [The Contribution Guidelines](https://github.com/reactjs/ar.reactjs.org#contributing), pick a page to translate and open a PR so I can assign you to it. enjoy!
> > الله حيو النشمي 😄
> 
> Thanks man @asantarissy , I have read them all just a while ago. I'm going to pick 'rendering elements' if I may.
> الله يحييك 😁

That's great @omar-dulaimi, please create a PR so I can assign you to this page.

**Comment by omar-dulaimi** [comment]
> > > > Can I join you guys please?
> > > > BTW "unit testing" = "اختبار الوحدات"
> > > 
> > > 
> > > Thanks @omar-dulaimi, please read [The Contribution Guidelines](https://github.com/reactjs/ar.reactjs.org#contributing), pick a page to translate and open a PR so I can assign you to it. enjoy!
> > > الله حيو النشمي 😄
> > 
> > 
> > Thanks man @asantarissy , I have read them all just a while ago. I'm going to pick 'rendering elements' if I may.
> > الله يحييك 😁
> 
> That's great @omar-dulaimi, please create a PR so I can assign you to this page.

I just did @asantarissy Thank you.

**Comment by aladin002dz** [comment]
Salam, Hello everyone,
I think we need to create a **facebook group** for all participants, for discussions, collaborations and sharing tips and experiences, and lead the project to a brilliant result.
Cheers,
Mahfoudh Arous

**Comment by aladin002dz** [comment]
> @fawzisf I'll look into it! For now, maybe looking at what the Persian translation is doing might be useful: https://github.com/reactjs/fa.reactjs.org/pull/7/files
indeed, I found some useful markdown tags in the project like these:
<h1 dir="rtl">
<p dir="rtl">

**Comment by aladin002dz** [comment]
> @fawzisf I'll look into it! For now, maybe looking at what the Persian translation is doing might be useful: https://github.com/reactjs/fa.reactjs.org/pull/7/files

indeed, I found some useful markdown tags in the project like these:
<h1 dir="rtl">
<p dir="rtl">
...

**Comment by asantarissy** [comment]
> Salam, Hello everyone,
> I think we need to create a **facebook group** for all participants, for discussions, collaborations and sharing tips and experiences, and lead the project to a brilliant result.
> Cheers,
> Mahfoudh Arous

Thanks @aladin002dz, I sent you a link on messenger to join us on slack.
Also, please check this #8 merged PR for the RTL implementation.

Enjoy.

**Comment by splimter** [comment]
i'll take **Basics** Lists and Keys

**Comment by asantarissy** [comment]
> i'll take **Basics** Lists and Keys

That's great @splimter, please create a PR so others won't work on the same page.

Enjoy!

**Comment by a-deeb** [comment]
Hey guys, can I help translate the  \repos\ar.reactjs.org\content\docs\Composition-vs-Inheritance page?

**Comment by ahmedfarag36** [comment]
Hello _good_ enthusiastic people;
This idea of improving the Arabic content never left my head so I am so happy to help,
I have been learning react for a while and read many of docs pages.
Ready to translate the tutorial or thinking in react 👍

**Comment by asantarissy** [comment]
> Hey guys, can I help translate the \ar.reactjs.org\content\docs\Composition-vs-Inheritance page?

Sure @a-deeb, please do the following:
* [ ] Read this issue
* [ ] Check our [Glossary](https://github.com/reactjs/ar.reactjs.org/wiki/Glossary) and [Style Guide](https://github.com/reactjs/reactjs.org-translation/blob/master/style-guide.md)
* [ ] Pick a page to translate
* [ ] Create a [PR](https://github.com/reactjs/ar.reactjs.org/pulls)
* [ ] And enjoy! 😄

**Comment by asantarissy** [comment]
> Hello _good_ enthusiastic people;
> This idea of improving the Arabic content never left my head so I am so happy to help,
> I have been learning react for a while and read many of docs pages.
> Ready to translate:
> 
> * the tutorial or
> * thinking in react 👍.

@ahmedfarag36 and we would be more than happy to have you on board. 
Read this issue and check our [Glossary](https://github.com/reactjs/ar.reactjs.org/wiki/Glossary) and [Style Guide](https://github.com/reactjs/reactjs.org-translation/blob/master/style-guide.md), the tutorial page is already taken, maybe you can start on Thinking in React page.

Please create a PR so I can add you here.

Cheers!

**Comment by awnigharbia** [comment]
Hey 👋,
Am gonna take the `Introducing Hooks` page.

**Comment by awnigharbia** [comment]
Hey @asantarissy, I don't think `Hooks` word needs to be translated, since it have other definistions in arabic, So I'll keep it as it is in `English` without not confusing to the reader, so they'll know it's a name of a feature added to react and doesn't have a meaning in arabic.

**Comment by asantarissy** [comment]
> Hey @asantarissy, I don't think `Hooks` word needs to be translated, since it have other definistions in arabic, So I'll keep it as it is in `English` without no confusing to the reader, so they'll know it's a name of a feature added to react and doesn't have a meaning in arabic.

Yes I agree with you, and this is what the team suggested.

Great then, enjoy.

**Comment by hgareeballa** [comment]
Hi 
i want to contribute, can someone assign a section to me !?
or i get to chose !?

THanks

**Comment by asantarissy** [comment]
> Hi
> i want to contribute, can someone assign a section to me !?
> or i get to chose !?
> 
> THanks

Welcome @hgareeballa, you can pick a page, please create a PR so I can assign you to it, also don't forget to check our [Glossary](https://github.com/reactjs/ar.reactjs.org/wiki/Glossary) and [Style Guide](https://github.com/reactjs/reactjs.org-translation/blob/master/style-guide.md)

Enjoy

**Comment by hgareeballa** [comment]
@asantarissy 
i will take //
API Reference
  React
  React.Component

Thanks

**Comment by asantarissy** [comment]
> @asantarissy
> i will take //
> API Reference
> React
> React.Component
> 
> Thanks

Great @hgareeballa, please create a "work in progress" PR for this ([steps](https://github.com/reactjs/ar.reactjs.org#contributing)) and use [Hsoub](https://wiki.hsoub.com/React/react_api) as reference for this.

**Comment by fkanout** [comment]
Hi everyone, I would like to contribute ... well, I've already submitted my PR before seeing this issue 🤷🏼‍♂️ Could you mention this issue in the contribution guide, so no one could miss it?

**Comment by osamajandali** [comment]
Hello everyone! 👋 
I am gonna take `Hooks at a Glance`

**Comment by ahmedfarag36** [comment]
Hello maintainers where are you?
I have started to translate the glossary however there is another pull request but I didn't notice since it is available here in the check list.

**Comment by aladin002dz** [comment]
> Hello maintainers where are you?
> I have started to translate the glossary however there is another pull request but I didn't notice since it is available here in the check list.

@asantarissy is our moderator, please ping him on the dedicated slack channel

**Comment by asantarissy** [comment]
> Hi everyone, I would like to contribute ... well, I've already submitted my PR before seeing this issue 🤷🏼‍♂️ Could you mention this issue in the contribution guide, so no one could miss it?

Thank you @fkanout, your PRs are in review, great job 💪🏼

Feel free to pick another page so I can assign it to you in this issue.

**Comment by asantarissy** [comment]
> Hello maintainers where are you?
> I have started to translate the glossary however there is another pull request but I didn't notice since it is available here in the check list.

Sorry @ahmedfarag36 my bad 🙂

Please Pick another page so I can assign you to it here.

**Comment by fkanout** [comment]
> > Hi everyone, I would like to contribute ... well, I've already submitted my PR before seeing this issue 🤷🏼‍♂️ Could you mention this issue in the contribution guide, so no one could miss it?
> 
> Thank you @fkanout and [done](https://github.com/reactjs/ar.reactjs.org#steps) thanks to @iRayan7, your PRs are in review, great job 💪🏼
> 
> Feel free to pick another page so I can assign it to you in this issue.

@asantarissy that's great, thank you and @iRayan7 👌🏻

**Comment by mhmdAljefri** [comment]
Hey I wanna help here. I will take Get Started page @tesseralis

**Comment by asantarissy** [comment]
> Hey I wanna help here. I will take Get Started page @tesseralis

That's great @mhmdAljefri, please create a work-in-progress PR so I can assign the page to you, don't forget to check our [Glossary](https://github.com/reactjs/ar.reactjs.org/wiki/Glossary), [Contribution](https://github.com/reactjs/ar.reactjs.org#contributing) and [Style](https://github.com/reactjs/reactjs.org-translation/blob/master/style-guide.md) Guides

**Comment by ZolfekarAskarieh** [comment]
Hello everyone, I wanna start helping here and I am gonna start working on "State and Lifecycle"

**Comment by asantarissy** [comment]
> Hello everyone, I wanna start helping here and I am gonna start working on "State and Lifecycle"

Great @ZolfekarAskarieh, please create a work-in-progress pull request so I can assign the page to you, also please go through our [Glossary](https://github.com/reactjs/ar.reactjs.org/wiki/Glossary), [Contribution](https://github.com/reactjs/ar.reactjs.org#contributing) and [Style](https://github.com/reactjs/reactjs.org-translation/blob/master/style-guide.md) Guides

**Comment by aminebenkeroum** [comment]
I will take the Error Boundaries page!

**Comment by asantarissy** [comment]
> I will take the Error Boundaries page!

That's great @aminebenkeroum, I'll assign the page to you, don't forget to go through our [Glossary](https://github.com/reactjs/ar.reactjs.org/wiki/Glossary), [Contribution](https://github.com/reactjs/ar.reactjs.org#contributing) and [Style](https://github.com/reactjs/reactjs.org-translation/blob/master/style-guide.md) Guides

**Comment by iron2017** [comment]
CodeEditor

**Comment by asantarissy** [comment]
> CodeEditor

Thanks @iron2017, page assigned 🎉
Please open a work-in-progress PR and check our [Glossary](https://github.com/reactjs/ar.reactjs.org/wiki/Glossary), [Contribution](https://github.com/reactjs/ar.reactjs.org#contributing) and [Style](https://github.com/reactjs/reactjs.org-translation/blob/master/style-guide.md) Guides

**Comment by ahmedfarag36** [comment]
Hello friends, why @splimter is taking too many pages?
If we translate them in parallel it would be more efficient and more convenient for him.
Thanks

**Comment by splimter** [comment]
@ahmedfarag36 i had time for it, i want to contribute before i get busy with univ stuff.
you can at anytime translate other section :)

**Comment by ahmedfarag36** [comment]
Ok @splimter, I appreciate you efforts and now I got the point happy translating :smiley:

**Comment by splimter** [comment]
@ahmedfarag36 btw am not doing the hard part, thanks to our maintainers <3.

**Comment by maskeddeveloper** [comment]
I would like to help in the translation to Arabic, 
I found that no one has been assigned to FAQ so is it possible for me to help translating ? 
I can start with APIs and AJAX and proceed sequentially on each FAQ page :)

**Comment by iRayan7** [comment]
> I would like to help in the translation to Arabic,
> I found that no one has been assigned to FAQ so is it possible for me to help translating ?
> I can start with APIs and AJAX and proceed sequentially on each FAQ page :)

Sure!

Thanks, @maskeddeveloper! `AJAX and APIs` page is assigned to you 🎉
Please create a "work in progress" PR for this. Check the [contribution steps](https://github.com/reactjs/ar.reactjs.org#contributing) and use [Hsoub](https://wiki.hsoub.com/React/faq_ajax) as reference.

**Comment by m1m6** [comment]
Hey Guys,
I'd like to contribute, and would like to start with ```react-without-jsx```

**Comment by iRayan7** [comment]
> Hey Guys,
> I'd like to contribute, and would like to start with `react-without-jsx`

Hey @m1m6! just assigned `react-without-jsx` to you 🎉
Please create a "work in progress" PR for this. Check the [contribution steps](https://github.com/reactjs/ar.reactjs.org#contributing). Thanks.

**Comment by ahmedfarag36** [comment]
Hello everybody,
I will take higher order components
Is it ok @iRayan7

**Comment by iRayan7** [comment]
> Hello everybody,
> I will take higher order components
> Is it ok @iRayan7

Hi @ahmedfarag36, `Higher-Order Components` is assigned to you 🎉
Please create a "work in progress" PR for this. Check the [contribution steps](https://github.com/reactjs/ar.reactjs.org#contributing). Thanks.

**Comment by AbdelkhalekESI** [comment]
Hey Guys,
I'd like to contribute, and would like to start with `Typechecking With PropTypes`

**Comment by AbdelkhalekESI** [comment]
Hi, 
I have done with the translation of `typechecking-with-proptypes.md` 
PR :  #80

**Comment by AbdelkhalekESI** [comment]
Hi Guys , 
So i'd like to contribute more and more in this translation so i will work this time on FAQ Section 
- `Babel, JSX, and Build Steps`
- `Passing Functions to Components`
I know that you recommend translate just one page but the first one is short . 
Thanks

**Comment by iRayan7** [comment]
> Hi Guys ,
> So i'd like to contribute more and more in this translation so i will work this time on FAQ Section
> 
> * `Babel, JSX, and Build Steps`
> * `Passing Functions to Components`
>   I know that you recommend translate just one page but the first one is short .
>   Thanks

Great @AbdelkhalekESI!
Please make a pull request for each. Thanks.

**Comment by maskeddeveloper** [comment]
Hello , I've done the faq-ajax, 
can i proceed to : 
- Component State
- Styling and CSS

Many Thanx

**Comment by iRayan7** [comment]
> Hello , I've done the faq-ajax,
> can i proceed to :
> 
> * Component State
> * Styling and CSS
> 
> Many Thanx

Assigned to you 👍 
Please create a draft PR for each one. Thanks.

**Comment by maskeddeveloper** [comment]
@iRayan7 i've done the 2 pages, please check it , could you please assign me the rest of the FAQ pages : 

-   File Structure
-   Versioning Policy
-   Virtual DOM and Internals

Many Thanx

**Comment by iRayan7** [comment]
> @iRayan7 i've done the 2 pages, please check it , could you please assign me the rest of the FAQ pages :
> 
> * File Structure
> * Versioning Policy
> * Virtual DOM and Internals
> 
> Many Thanx

Go for it! 😍 
Please create a draft PR for each. Thanks.

**Comment by 3imed-jaberi** [comment]
Hello Guys,
I would like to help in the translation to Arabic,
I will start with ' Static Type Checking ' page .

**Comment by aladin002dz** [comment]
> Hello Guys,
> I would like to help in the translation to Arabic,
> I will start with ' Static Type Checking ' page .

Hey! Okay, you can fork, clone create another branch on which you change the title of the the page you booked, and then send the pull request,
please, use this translation:
https://wiki.hsoub.com/React
Happy working,

**Comment by 3imed-jaberi** [comment]
I'll take 'strict-mode' page .

**Comment by Fcmam5** [comment]
Hello!
I would like to help :smile: , may I start with `Optimizing Performance` ?

**Comment by 3imed-jaberi** [comment]
Welcome @Fcmam5 .. 
Okey , you can take this page and please , make sure you use  this translation: https://wiki.hsoub.com/React 
and read the Gloassary : 
https://github.com/reactjs/ar.reactjs.org/wiki/Glossary ..


Happy working

**Comment by 3imed-jaberi** [comment]
I'll take 'web-components' and 'uncontrolled-components' pages .

**Comment by 3imed-jaberi** [comment]
I'll take 'docs/nav.yml'.

**Comment by Fcmam5** [comment]
Hello everyone,
I am finished with **Optimizing Performance** (#96).
Can I proceed to `Reconciliation`?

And how about the pages that have been assigned but they became inactive like (#4, #7, #10 ...), what can we do about it?

**Comment by iRayan7** [comment]
> Hello everyone,
> I am finished with **Optimizing Performance** (#96).
> Can I proceed to `Reconciliation`?
> 
> And how about the pages that have been assigned but they became inactive like (#4, #7, #10 ...), what can we do about it?

Thanks, @Fcmam5! 
Sure `Reconciliation` has been assigned to you.

For the inactive pages, we will re-assign them once we complete the other unassigned pages.

**Comment by Fcmam5** [comment]
As Salam Alikum,
I finished translating **Reconciliation** (#99), and I'd like to continue with:
* Portals
* Render Props

And for sure each of them will be in a seprate PR :nerd_face:

**Comment by 3imed-jaberi** [comment]
@Fcmam5 sorry bro , i finish 'Portals' page and i sended a PR 😕 ..

**Comment by Fcmam5** [comment]
> @Fcmam5 sorry bro , i finish 'Portals' page and i sended a PR confused ..

No worries :) thank you for your contribution :smile: !

**Comment by 3imed-jaberi** [comment]
> > @Fcmam5 sorry bro , i finish 'Portals' page and i sended a PR 😕  ..
> 
> No worries :) thank you for your contribution 😄 !

Thank you too for your efforts and contribution 🤝

**Comment by 3imed-jaberi** [comment]
I'll take 'Integrating with Other Libraries' page.

**Comment by 3imed-jaberi** [comment]
I'll take 'JSX in depth' page too 🤗.

**Comment by mohshbool** [comment]
I'll take 'Refs and the DOM' page

**Comment by 3imed-jaberi** [comment]
@tesseralis  @22mahmoud  @Emmanuel-Melon  @iRayan7 @asantarissy @aladin002dz 
Please make sure that the people who took the pages and didn't send PR are still contributing 🤔 ‼️ ...
In case of abandonment, please update the list 😁.

**Comment by Fcmam5** [comment]
I would like to work on the components section:
- CodeEditor (it's not active anymore)
- CodeExample
- ErrorDecoder
- LayoutFooter
- LayoutHeader
- MarkdownPage

**Comment by 3imed-jaberi** [comment]
> I would like to work on the components section:
> 
> * CodeEditor (it's not active anymore)
> * CodeExample
> * ErrorDecoder
> * LayoutFooter
> * LayoutHeader
> * MarkdownPage
> 
> Changes will only affect just a few lines in:
> 
> ```
> src/components/CodeEditor/CodeEditor.js
> src/components/CodeExample/CodeExample.js
> src/components/ErrorDecoder/ErrorDecoder.js
> src/components/LayoutHeader/Header.js
> src/components/LayoutHeader/SearchSvg.js
> ```
> 
> Should it be only in one pull request?

### Bro , if you needs help i'm with you in this section.. just tell me ❤️🤝🔥

**Comment by mohshbool** [comment]
Just finished refs and the dom (#108). Not sure what's left since the list above isn't, afaik, up-to-date

**Comment by iRayan7** [comment]
> I would like to work on the components section:
> 
> * CodeEditor (it's not active anymore)
> * CodeExample
> * ErrorDecoder
> * LayoutFooter
> * LayoutHeader
> * MarkdownPage
> 
> Changes will only affect just a few lines in:
> 
> ```
> src/components/CodeEditor/CodeEditor.js
> src/components/CodeExample/CodeExample.js
> src/components/ErrorDecoder/ErrorDecoder.js
> src/components/LayoutHeader/Header.js
> src/components/LayoutHeader/SearchSvg.js
> ```
> 
> Should it be only in one pull request?

Yes, please. One PR would be fine 👍

**Comment by Fcmam5** [comment]
I will work on:
* How to Contribute
* Community

**Comment by mohshbool** [comment]
I would like to take Virtual DOM and Internals if it's not taken by @maskeddeveloper since there's a strikethrough

**Comment by mohshbool** [comment]
I'll also take Warnings. I'll be submitting each file in a pull request of its own

**Comment by aladin002dz** [comment]
is there any page left, I can help with?

**Comment by iRayan7** [comment]
> I would like to take Virtual DOM and Internals if it's not taken by @maskeddeveloper since there's a strikethrough

It's already been translated by @3imed-jaberi

**Comment by iRayan7** [comment]
Hey guys 👋
Please stick with the process of page assignment. You **have to** comment before you start working on it.
Thanks ❤️

**Comment by iRayan7** [comment]
These pages are available:

#### Basics
* [ ] Hello World
#### Advanced Guides
* [ ] Higher-Order Components

Please comment before you start working on it 👍

**Comment by maskeddeveloper** [comment]
@iRayan7 assign me the  Hello World Page

**Comment by aladin002dz** [comment]
@iRayan7, assign me the "Higher-Order Components" page

**Comment by atefBB** [comment]
@tesseralis @iRayan7 as discussed https://github.com/reactjs/ar.reactjs.org/pull/4,
@Emmanuel-Melon still working on the PR !

**Comment by iRayan7** [comment]
> @iRayan7 assign me the Hello World Page

> @iRayan7, assign me the "Higher-Order Components" page

Sure, 👍Open up a new PR, please.

**Comment by Emmanuel-Melon** [comment]
> @tesseralis @iRayan7 as discussed [here](https://github.com/reactjs/ar.reactjs.org/pull/4),
> @Emmanuel-Melon still working on the PR !

@atefBB, it's okay, you could work on it. Would you like to do a complete do over or pick up from where I'd left off?

**Comment by ahmedkrmn** [comment]
@iRayan7 Hey Rayan, can I be assigned to translate these remaining warning messages:
`legacy-factories`, `refs-must-have-owner`, `special-props`, and `unknown-prop` ?

**Comment by mohshbool** [comment]
> @iRayan7 Hey Rayan, can I be assigned to translate these remaining warning messages:
> [`legacy-factories`](https://github.com/reactjs/ar.reactjs.org/blob/master/content/warnings/legacy-factories.md), [`refs-must-have-owner`](https://github.com/reactjs/ar.reactjs.org/blob/master/content/warnings/refs-must-have-owner.md), [`special-props`](https://github.com/reactjs/ar.reactjs.org/blob/master/content/warnings/special-props.md), and [`unknown-prop`](https://github.com/reactjs/ar.reactjs.org/blob/master/content/warnings/unknown-prop.md) ?

I'm sorry Ahmed but I'm already doing them I just haven't submitted the PRs yet. How about checking the 'Contributing' section?

**Comment by ahmedkrmn** [comment]
> > @iRayan7 Hey Rayan, can I be assigned to translate these remaining warning messages:
> > [`legacy-factories`](https://github.com/reactjs/ar.reactjs.org/blob/master/content/warnings/legacy-factories.md), [`refs-must-have-owner`](https://github.com/reactjs/ar.reactjs.org/blob/master/content/warnings/refs-must-have-owner.md), [`special-props`](https://github.com/reactjs/ar.reactjs.org/blob/master/content/warnings/special-props.md), and [`unknown-prop`](https://github.com/reactjs/ar.reactjs.org/blob/master/content/warnings/unknown-prop.md) ?
> 
> I'm sorry Ahmed but I'm already doing them I just haven't submitted the PRs yet. How about checking the 'Contributing' section?

I saw you did `dont-call-proptypes`, `invalid-aria-prop`, and `invalid-hook-call-warning`, but those I am asking for aren't assigned to anyone!

**Comment by mohshbool** [comment]
> > > @iRayan7 Hey Rayan, can I be assigned to translate these remaining warning messages:
> > > [`legacy-factories`](https://github.com/reactjs/ar.reactjs.org/blob/master/content/warnings/legacy-factories.md), [`refs-must-have-owner`](https://github.com/reactjs/ar.reactjs.org/blob/master/content/warnings/refs-must-have-owner.md), [`special-props`](https://github.com/reactjs/ar.reactjs.org/blob/master/content/warnings/special-props.md), and [`unknown-prop`](https://github.com/reactjs/ar.reactjs.org/blob/master/content/warnings/unknown-prop.md) ?
> > 
> > 
> > I'm sorry Ahmed but I'm already doing them I just haven't submitted the PRs yet. How about checking the 'Contributing' section?
> 
> I saw you did `dont-call-proptypes`, `invalid-aria-prop`, and `invalid-hook-call-warning`, but those I am asking for aren't assigned to anyone!

I know but I'm doing them as I said in a [comment above](https://github.com/reactjs/ar.reactjs.org/issues/1#issuecomment-506274766)

**Comment by ahmedkrmn** [comment]
@mohshbool sorry Mohammad, haven't seen your comment.
@iRayan7 Can I take the [Codebase Overview](https://github.com/reactjs/ar.reactjs.org/blob/master/content/docs/codebase-overview.md) page?

**Comment by mohshbool** [comment]
@ahmedkrmn all is well 😅
You should be able to take "codebase overview", we would be more than happy to have you on board.
Read this issue and check our [Glossary](https://github.com/reactjs/ar.reactjs.org/wiki/Glossary) and [Style Guide](https://github.com/reactjs/reactjs.org-translation/blob/master/style-guide.md).

Please create a PR so I can add you here.

**Comment by ahmedkrmn** [comment]
> @ahmedkrmn all is well 😅
> You should be able to take "codebase overview", we would be more than happy to have you on board.
> Read this issue and check our [Glossary](https://github.com/reactjs/ar.reactjs.org/wiki/Glossary) and [Style Guide](https://github.com/reactjs/reactjs.org-translation/blob/master/style-guide.md).
> 
> Please create a PR so I can add you here.

Thanks. The PR is #131.

**Comment by Aissaoui-Ahmed** [comment]
Hi !
any page available

**Comment by mohshbool** [comment]
> Hi !
> any page available

hey @Aissaoui-Ahmed, you might wanna check out the 'Contributing' section or perhaps the Community issue #117

**Comment by Aissaoui-Ahmed** [comment]
thank you @mohshbool

**Comment by Fcmam5** [comment]
I will take `Implementation Notes`

**Comment by mohdule** [comment]
I will translate the `Testing Overview` page

**Comment by ahmdtalaat** [comment]
Half of the word "مجتمع المطورين"  is hiding behind the search bar, Would like to work on that

**Comment by ahmdtalaat** [comment]
Hello Guys, so enthusiastic to begin, Shall We refactor Todo app in the tutorial to Function components (Hooks) ?

**Comment by ahmdtalaat** [comment]
Anyone is working on /languages page ??

**Comment by 3imed-jaberi** [comment]
@mohdule , you can take `Testing Overview` page.

**Comment by 3imed-jaberi** [comment]
> Anyone is working on `/languages` page ?? Can I work on it?

Do you mean `languages.yml` file !!

**Comment by ahmdtalaat** [comment]
@3imed-jaberi I mean ar.reactjs.org/languages

**Comment by 3imed-jaberi** [comment]
> @3imed-jaberi I mean ar.reactjs.org/languages

🚀 ..

**Comment by mohdule** [comment]
Alright cool @3imed-jaberi, gonna get it done by next Sat hopefully

**Comment by mohamedsgap** [comment]
Hi @tesseralis and @3imed-jaberi I really want to contribute and translate any of the Testing ( New ) pages, please assign to me any page and I will translate it :)

**Comment by mohamedsgap** [comment]
Is there  any opportunity to contribute in the translation and improve it ? @tesseralis

**Comment by 3imed-jaberi** [comment]
@mohamedsgap ,  welcome 👋 !
It's good to meet enthusiastic people like you ❤️ 
You can start with ` Testing Recipes`  🚀 ..

**Comment by forabi** [comment]
Hey everyone, do you have plans to translate the new concurrent mode pages? I can take a stab at this!

**Comment by cdes** [comment]
I can help too 🌚

**Comment by 3imed-jaberi** [comment]
@forabi @cdes 👋 ..
I would like to thank you for your interest .. 
We are currently working to integrate (merge) all updates to our Arabic content and soon this section will be ready ...
Now, you can start with `Testing Environments` or `Design Principles`. 
You can also help @Fcmam5 in #163 ...

**Comment by Karamiq** [comment]
> Can I join you guys please?
> BTW "unit testing" = "اختبار الوحدة"

**Comment by 3imed-jaberi** [comment]
@Karamiq welcome 👋 ..
you can start with `Testing Recipes`  or ` Testing Environments`  🚀 ..

**Comment by mohamedsgap** [comment]
> @Karamiq welcome 👋 ..
> you can start with `Testing Recipes` or ` Testing Environments` 🚀 ..

@3imed-jaberi  
I'm currently working on Testing Recipes

**Comment by 3imed-jaberi** [comment]
> @Karamiq welcome 👋 ..
> ~you can start with `Testing Recipes` or ` Testing Environments` 🚀 ..~

@Karamiq , you can start with ` Testing Environments` 🚀 ..

**Comment by cdes** [comment]
I'm working on `Design Principles`

**Comment by mohamedsgap** [comment]
@3imed-jaberi   @Aissaoui-Ahmed  I gonna work on Testing Environments!

**Comment by mohamedsgap** [comment]
@3imed-jaberi  @Aissaoui-Ahmed @asantarissy  Hi guys! I finished translating Testing Environmnts, please review my pull request :)

**Comment by asantarissy** [comment]
> @3imed-jaberi I'm working on `Design Principles`

@cdes what's the status on this? 😃

**Comment by mohamedsgap** [comment]
@3imed-jaberi  @Aissaoui-Ahmed  @tesseralis  @asantarissy  Hey guys :) I wonder if there ara any opporunities to contribute to more repos related to React to write a real code this time! or any available pages i noticed that Concurrent Mode ( New ) is created, one more thing how do I become one of React Community like you :)

**Comment by mohamedsgap** [comment]
Hey guys! @asantarissy @iRayan7 @aladin002dz @Fcmam5 @mohshbool @3imed-jaberi @splimter please assign to me any page to translate it!

**Comment by asantarissy** [comment]
> Hey guys! @asantarissy @iRayan7 @aladin002dz @Fcmam5 @mohshbool @3imed-jaberi @splimter please assign to me any page to translate it!

Thanks @mohamedsgap, please pick a page/s and create a PR so I can assign you.

Enjoy 😊

**Comment by mohamedsgap** [comment]
Hey guys! @asantarissy @iRayan7 @aladin002dz @Fcmam5 @mohshbool @3imed-jaberi @splimter I'm going to translate Release Channels!

**Comment by Fcmam5** [comment]
> Hey guys! @asantarissy @iRayan7 @aladin002dz @Fcmam5 @mohshbool @3imed-jaberi @splimter I'm going to translate Release Channels!

Good luck Mohamed 🚀 !

**Comment by Fcmam5** [comment]
I will take "Introducing Concurrent Mode"

**Comment by mohamedsgap** [comment]
I gonna work on ''Suspense for Data Fetching''  @asantarissy @iRayan7 @aladin002dz @Fcmam5 @mohshbool @3imed-jaberi @splimter  @Aissaoui-Ahmed

**Comment by cdes** [comment]
> > @3imed-jaberi I'm working on `Design Principles`
> 
> @cdes what's the status on this? 😃

I'm unable to work on it. Sorry.

**Comment by mohamedsgap** [comment]
I've just finished translated the most challenging page I've ever translated "Suspense for data fetching", I need your review, guys! :)
@asantarissy @iRayan7 @aladin002dz @Fcmam5 @mohshbool @3imed-jaberi @splimter @Aissaoui-Ahmed

**Comment by HassanMuhamad** [comment]
I'm working on " Concurrent UI Patterns" documentation. ⌨️

**Comment by mohamedsgap** [comment]
@HassanMuhamad great choice, Hassan! keep going and I will follow you up with  'Adopting Concurrent Mode' after my current PR's been merged!

**Comment by hassan** [comment]
The line above: `Concurrent UI Patterns (@hassan Muhamad)` incorrectly tags me; can the author correct? Sorry, I don't speak Arabic 😞

**Comment by asantarissy** [comment]
> The line above: `Concurrent UI Patterns (@hassan Muhamad)` incorrectly tags me; can the author correct? Sorry, I don't speak Arabic 😞

Thanks @hassan for the heads-up 😊

**Comment by heithemmoumni** [comment]
I will take "Implementation Notes" and "Design Principles"

**Comment by heithemmoumni** [comment]
I will take "Implementation Notes"

**Comment by atefBB** [comment]
Welcome to @all newcomers!

**Comment by mohamedsgap** [comment]
Hi guys,
@3imed-jaberi, @Aissaoui-Ahmed, @aladin002dz, @asantarissy and @Fcmam5
just finished translating 'Adopting Concurrent Mode' I need your review :)

**Comment by mohamedsgap** [comment]
I gonna translate a new page "Concurrent Mode API Reference" @tesseralis

**Comment by YassinHussein** [comment]
Hi guys, I would like to translate "Concurrent UI Patterns" page

**Comment by asantarissy** [comment]
> Hi guys, I would like to translate "Concurrent UI Patterns" page

Thank you for joining the team @YassinHussein, please go ahead I'll assign the page to you.

* [ ] Read this issue
* [ ] Check our [Glossary](https://github.com/reactjs/ar.reactjs.org/wiki/Glossary) and [Style Guide](https://github.com/reactjs/reactjs.org-translation/blob/master/style-guide.md)
* [x] Pick a page to translate
* [ ] Create a [PR](https://github.com/reactjs/ar.reactjs.org/pulls)
* [ ] And enjoy! 😄

**Comment by maqalaqil** [comment]
I would love to take a page please assign me whatever is available   Versioning Policy || profiler API

**Comment by 3imed-jaberi** [comment]
Welcome @maqalaqil 👋🏻, it's yours.

**Comment by maqalaqil** [comment]
Thank you :D

**Comment by tech-chieftain** [comment]
Hey all, would love to help you guys with this. Can you assign Implementation Notes & Design Principles to me?

**Comment by mriiad** [comment]
Hi guys,
I would like to translate 'Implementation Notes' page, would it be possible to assign it to me please ?
@3imed-jaberi @Aissaoui-Ahmed, @aladin002dz, @asantarissy, @Fcmam5 
Thank you!

**Comment by 3imed-jaberi** [comment]
Hello @mriiad, Sure you can take it. You need to submit a PR.

**Comment by 3imed-jaberi** [comment]
Hello @tech-chieftain @maqalaqil @YassinHussein,

Are there any updates about the pages you took?

**Comment by mriiad** [comment]
> Hello @mriiad, Sure you can take it. You need to submit a PR.

Hello again @3imed-jaberi 

Thanks for the quick reply, here's the PR associated to this task https://github.com/reactjs/ar.reactjs.org/pull/372

Could you just add all the suitable labels and assign yourself as a reviewer please ?

Thank you!

**Comment by tech-chieftain** [comment]
Hey @3imed-jaberi, as Implementation Notes was assigned to @mriiad I can still work on Design Principles. Can you please assign it to me so others can focus on other pages? It will be ready in the next couple of days. Thanks

**Comment by maqalaqil** [comment]
Hey @3imed-jaberi ill have the PR ready today.

**Comment by ahmed-abdulhuy** [comment]
Hi guys,
I would like to translate the 'how to contribute' [#branch-organization](https://ar.reactjs.org/docs/how-to-contribute.html#branch-organization) & [#feature-flags](https://ar.reactjs.org/docs/how-to-contribute.html#feature-flags) page would it be possible to assign it to me, please?
Thank you!

**Comment by Aissaoui-Ahmed** [comment]
@ahmed-abdulhuy you can open PR
Thank you for your contribution

**Comment by ahmed-abdulhuy** [comment]
I would like to translate the 'Profiler API'  page would it be possible to assign it to me, please?
Thank you!

**Comment by 3imed-jaberi** [comment]
@tech-chieftain @maqalaqil, if you take any page please open a draft pr to help me to assign it to you!

**Comment by Sebastianoskiojoj** [comment]
b1-2283a18c-b2a2-4dd2-ac5c-d4563dd8ce46

**Comment by FettahAud** [comment]
Hey, I would like to translate "Design Principles" in **Contributing** section
Which is not assigned to anyone as i can see in the list

**Comment by yusuf963** [comment]
Hi, I will take `Concurrent UI Patterns` - thanks

**Comment by yusuf963** [comment]
Hi, I will take `Concurrent UI Patterns` - thanks

**Comment by yusuf963** [comment]
also I did an initial start and translate the head of concurrent-mode-patterns.md file in this draft https://github.com/reactjs/ar.reactjs.org/pull/454

**Comment by 3imed-jaberi** [comment]
Hello guys, first let's say RAMADHAN MOUBAREK ☪︎❤️.

I would say, in the coming days, we will start the translation for the new website version.
For that, I will create a new progress-tracking issue, and in the meanwhile, the missing pages here are still available for translation.

**Comment by gaearon** [comment]
I created https://github.com/reactjs/ar.react.dev/issues/464 for the new translation. Let's coordinate there!
